docs/how-to/craft/configure-distribution-information.rst

```diff
@@ -0,0 +1,351 @@
+main part.
+
+After that, set ``override-pull`` to an inline series of ``craftctl`` commands. The
+variables ``version`` and ``grade`` map to the keys with the same names. You can set
+them like environment variables with the ``set`` verb. Here you can use any external
+source that's accessible through terminal commands from the host's environment, such as an environment variable
+variable or an API endpoint.
+
+During build, Snapcraft will now set the snap's version and grade based on the values
+from any source you like.
+
+Here's an example of that configuration:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+    :emphasize-lines: 2, 9-12
+
+    name: bravo
+    adopt-info: bravo-part
+    ...
+
+    parts:
+      bravo-part:
+        plugin: dump
+        source: http://github.com/alex/bravo.git
+        override-pull: |
+          craftctl default
+          craftctl set version="1.5.3"
+          craftctl set grade="stable"
+
+
+Configure the desktop entry
+---------------------------
+
+Snaps support the Linux `desktop entry
+<https://specifications.freedesktop.org/desktop-entry-spec/latest>` standard. You can
+use desktop entry files to define your snap's entry in the desktop environment's various
+app menus and launchers. The file controls the entry's presentation and how it launches.
+If configured, snapd will automatically add your snap to the app launcher and menus
+during installation.
+
+There are three methods to configure the desktop menu entry:
+
+- Copy the desktop entry file to the ``snap/gui`` directory.
+- Point the ``desktop`` key in the app definition to a desktop file in the ``prime``
+  directory.
+- `Use the desktop entry file
+  <how-to-configure-distribution-information-appstream-metadata>`_ from the AppStream
+  metadata of your app.
+
+.. important::
+
+    The icon in the desktop entry is separate from the ``icon`` key. The latter is used
+    in graphical front ends, like the snap's profile in the Snap Store.
+
+
+Add a desktop entry file
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To start, create files named ``<snap-name>.desktop`` and ``<snap-name>.png`` in the
+``snap/gui/`` directory in your project's source. Replace ``<snap-name>`` with the same
+value as the ``name`` key in the project file.
+
+For the desktop entry file, enter:
+
+.. code-block:: desktop
+    :caption: .desktop file
+
+    [Desktop Entry]
+    Exec=<app-name>
+    Icon=${SNAP}/meta/gui/<snap-name>.png
+
+Replace ``<app-name>`` with the same name you gave the app in the project file. The
+name is case-sensitive.
+
+Assign ``Icon`` to the absolute path of the image file. This path must be the location
+of the icon after the snap is installed. Since snapcraft copies all the contents of the
+``snap/gui/`` folder to ``meta/gui`` in the installation, the absolute path of the icon
+in this arrangement is ``${SNAP}/meta/gui/snapname.png``.
+
+
+Read a desktop entry file
```

Comment:
I sense that this might be the most common or desireable case. Should I make it the first one?

Reply:
df0d47c